kratos/utilities/exact_mortar_segmentation_utility.h

```diff
@@ -500,87 +513,143 @@ class KRATOS_API(KRATOS_CORE) ExactMortarIntegrationUtility
 
     /**
      * @brief This function push backs the points that are inside
-     * @param PointList The intersection points
-     * @param AllInside The nodes that are already known as inside the other geometry
-     * @param ThisGeometry The geometry considered
+     * @param rPointList The intersection points
+     * @param rAllInside The nodes that are already known as inside the other geometry
+     * @param rThisGeometry The geometry considered
      */
+    template<SizeType TSizeCheck = TNumNodes>
     inline void PushBackPoints(
-        VectorPoints& PointList,
-        const array_1d<bool, TNumNodes>& AllInside,
-        GeometryPointType& ThisGeometry
-        );
+        VectorPoints& rPointList,
+        const array_1d<bool, TSizeCheck>& rAllInside,
+        GeometryPointType& rThisGeometry
+        )
+    {
+        for (IndexType i_node = 0; i_node < TSizeCheck; ++i_node) {
+            if (rAllInside[i_node]) {
+                // We check if the node already exists
+                bool add_point = true;
```

Comment:
I think the following would be easier and more efficient (bcs of the break):
~~~
for (IndexType iter = 0; iter < rPointList.size(); ++iter) {
    if (CheckPoints(rThisGeometry[i_node], rPointList[iter])) {
        rPointList.push_back(rThisGeometry[i_node]);
        break;
    }
}
~~~

Reply:
I will check

Reply:
Doesn't work, is the opposite, we consider true by default, and we check if we don't add